__tests__/__unit__/dataset/actions.unit.test.ts

```diff
@@ -0,0 +1,1508 @@
+            commandResponse: "success",
+            apiResponse: [{
+                etag: "123"
+            }]
+        });
+        mocked(vscode.window.withProgress).mockImplementation((progLocation, callback) => {
+            return callback();
+        });
+        blockMocks.profileInstance.loadNamedProfile.mockReturnValueOnce(blockMocks.imperativeProfile);
+        mocked(Profiles.getInstance).mockReturnValue(blockMocks.profileInstance);
+        const mockSetEtag = jest.spyOn(node, "setEtag").mockImplementation(() => null);
+        const testDocument = generateTextDocument(blockMocks.datasetSessionNode, "HLQ.TEST.AFILE");
+        (testDocument as any).fileName = path.join(globals.DS_DIR, "Favorites", testDocument.fileName);
+
+        await dsActions.saveFile(testDocument, blockMocks.testDatasetTree);
+
+        expect(mocked(sharedUtils.concatChildNodes)).toBeCalled();
+        expect(mockSetEtag).toHaveBeenCalledWith("123");
+        expect(mocked(vscode.window.showInformationMessage)).toBeCalledWith("success");
+        expect(blockMocks.profileInstance.loadNamedProfile).toBeCalledWith("TestSessionName");
+    });
+    it("Checking favorite PDS Member saving", async () => {
+        globals.defineGlobals("");
+        createGlobalMocks();
+        blockMocks = createBlockMocks();
+        const node = new ZoweDatasetNode("[TestSessionName]: HLQ.TEST.AFILE", vscode.TreeItemCollapsibleState.None,
+            blockMocks.datasetSessionNode, null, undefined, undefined, blockMocks.imperativeProfile);
+        node.contextValue = globals.DS_PDS_CONTEXT;
+        const childNode = new ZoweDatasetNode("MEM", vscode.TreeItemCollapsibleState.None,
+            node, null, undefined, undefined, blockMocks.imperativeProfile);
+        const favoriteNode = new ZoweDatasetNode("[TestSessionName]: HLQ.TEST.AFILE", vscode.TreeItemCollapsibleState.None,
+            blockMocks.datasetSessionNode, null, undefined, undefined, blockMocks.imperativeProfile);
+        favoriteNode.contextValue = globals.DS_PDS_CONTEXT + globals.FAV_SUFFIX;
+        const favoriteChildNode = new ZoweDatasetNode("MEM", vscode.TreeItemCollapsibleState.None,
+            favoriteNode, null, undefined, undefined, blockMocks.imperativeProfile);
+        node.children.push(childNode);
+        favoriteNode.children.push(favoriteChildNode);
+        blockMocks.datasetSessionNode.children.push(node);
+        blockMocks.testDatasetTree.mFavorites.push(favoriteNode);
+
+        mocked(sharedUtils.concatChildNodes).mockReturnValueOnce([node, childNode]);
+        blockMocks.testDatasetTree.getChildren.mockReturnValueOnce([blockMocks.datasetSessionNode]);
+        mocked(zowe.List.dataSet).mockResolvedValue({
+            success: true,
+            commandResponse: "",
+            apiResponse: {
+                items: [{ dsname: "HLQ.TEST.AFILE" }, { dsname: "HLQ.TEST.AFILE(MEM)" }]
+            }
+        });
+        mocked(zowe.Upload.pathToDataSet).mockResolvedValueOnce({
```

Comment:
There's a mockCreator function for this, you can pass in the apiResponse as a parameter.

Also does the mock resolved value need to be assigned in every it() block? Maybe this can be defined in createBlockMocks()?